stats-test.cpp

```diff
@@ -2,22 +2,32 @@
 
 #include "catch.hpp"
 #include "stats.h"
-
 #include <cmath>
+#include <math.h>
 
-TEST_CASE("reports average, minimum and maximum") {
+TEST_CASE("when all inputs are real numbers of type float then report average, minimum and maximum to an accuracy of 0.001") {
     auto computedStats = Statistics::ComputeStatistics({1.5, 8.9, 3.2, 4.5});
-    float epsilon = 0.001;
+    float epsilon = 0.001f;
     REQUIRE(std::abs(computedStats.average - 4.525) < epsilon);
     REQUIRE(std::abs(computedStats.max - 8.9) < epsilon);
     REQUIRE(std::abs(computedStats.min - 1.5) < epsilon);
 }
 
-TEST_CASE("average is NaN for empty array") {
+TEST_CASE("average, min, max is NaN for empty array") {
     auto computedStats = Statistics::ComputeStatistics({});
     //All fields of computedStats (average, max, min) must be
     //NAN (not-a-number), as defined in math.h
     
     //Design the REQUIRE statement here.
+    REQUIRE(isnan(computedStats.average)==true);
+    REQUIRE(isnan(computedStats.max)==true);
+    REQUIRE(isnan(computedStats.min)==true);
     //Use http://www.cplusplus.com/reference/cmath/isnan/
 }
+TEST_CASE("when some elements in the input are NaN then calculate stats of remaining numbers") {
```

Comment:
Nicely described test. Also good practice to take the vector by value in RemoveNAN, so that the original vector is not modified 👍

Reply:
Thank you 👍